core/include/detray/tools/single_type_navigator.hpp

```diff
@@ -272,15 +286,12 @@ class single_type_navigator {
 
         // Should never be the case after update call (without portals we are
         // trapped)
-        if (navigation.kernel().empty()) {
+        if (navigation.kernel().empty() and heartbeat) {
```

Comment:
Why are you killing it when heart is beating?

Reply:
Because if the first predicate is true and we are in the middle of navigation (i.e. heartbeat true), we need to abort. It could however also be the case, that we are trying to exit already and the kernel was cleared. In this case the heartbeat will be false, but we don't want to also abort.

Reply:
I see - but is it possible that `heartbeat` can be false here? It seems not changing after it is called in `status`

Reply:
Anyway doesnt seem to be a big problem